tests/tests/integration_tests.rs

```diff
@@ -393,34 +393,40 @@ async fn ws_server_cancels_stream_after_reset_conn() {
 	let server = WsServerBuilder::default().build("127.0.0.1:0").await.unwrap();
 	let server_url = format!("ws://{}", server.local_addr().unwrap());
 
-	let (tx, mut rx) = mpsc::channel(1);
+	let (tx, rx) = mpsc::channel(1);
 	let mut module = RpcModule::new(tx);
 
 	module
-		.register_subscription("subscribe_never_produce", "n", "unsubscribe_never_produce", |_, sink, mut tx| {
-			// create stream that doesn't produce items.
-			let stream = futures::stream::empty::<usize>();
+		.register_subscription("subscribe_for_ever", "n", "unsubscribe_for_ever", |_, sink, mut tx| {
+			// Create stream that produce one item then sleeps for an hour.
+			let interval = interval(Duration::from_secs(60 * 60));
+			let stream = IntervalStream::new(interval).map(move |_| 0_usize);
+
 			tokio::spawn(async move {
 				sink.pipe_from_stream(stream).await.unwrap();
 				let send_back = Arc::make_mut(&mut tx);
-				send_back.feed(()).await.unwrap();
+				send_back.send(()).await.unwrap();
 			});
+
 			Ok(())
 		})
 		.unwrap();
 
 	server.start(module).unwrap();
 
 	let client = WsClientBuilder::default().build(&server_url).await.unwrap();
-	let _sub1: Subscription<usize> =
-		client.subscribe("subscribe_never_produce", None, "unsubscribe_never_produce").await.unwrap();
-	let _sub2: Subscription<usize> =
-		client.subscribe("subscribe_never_produce", None, "unsubscribe_never_produce").await.unwrap();
+	let mut subs = Vec::new();
+
+	for _ in 0..10 {
+		subs.push(client.subscribe::<usize>("subscribe_for_ever", None, "unsubscribe_for_ever").await.unwrap());
+	}
 
 	// terminate connection.
 	drop(client);
-	assert_eq!(Some(()), rx.next().await, "subscription stream should be terminated after the client was dropped");
-	assert_eq!(Some(()), rx.next().await, "subscription stream should be terminated after the client was dropped");
+
+	let rx_len = rx.take(10).fold(0, |acc, _| async move { acc + 1 }).await;
```

Comment:
it could a few milliseconds to send the WS close message but so not really is my understanding `rx` here is another channel where `tx` is encapsulated in the `RpcModule` so it's not the `subscription stream`

Then after connection is closed the actual subscriptions should be terminated using `Notify::notify_waiters` after that this messages and sent to channel above...